controllers/machinehealthcheck_controller_test.go

```diff
@@ -110,14 +110,16 @@ var _ = Describe("MachineHealthCheck", func() {
 					nodes = append(nodes, node)
 
 					if nodeRef {
+						machinePatch, err := patch.NewHelper(machine, testEnv)
```

Comment:
Should we be relying on the patch helper for tests? It seems like the change we are making is relatively minor and the default controller runtime patching utilities should be sufficient for our needs.

Reply:
We could, any reason why we shouldn't use the helper?

Reply:
Updated, ptal

Reply:
> We could, any reason why we shouldn't use the helper?

Mainly out of concern that it is non-trivial code that could itself have bugs that could affect the outcome of the tests creating either false positives or false negatives.